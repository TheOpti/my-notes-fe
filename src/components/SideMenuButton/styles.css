.sideMenuButton {
	border-radius: 0 25px 25px 0;
	color: var(--main-gray-text-color);
	display: flex;
	align-items: center;
	line-height: 1.25rem;
	height: 48px;
	min-width: 48px;
	overflow: hidden;
	width: 100%;
	cursor: pointer;
}

.expanded {
	padding-left: 12px;
}

.collapsed {
	margin-left: 12px;
	border-radius: 25px;
}

.sideMenuButton:hover {
	background-color: #f1f3f4;
}

.active,
.active:hover {
	background-color: #feefc3;
}

.icon {
	width: 24px;
	margin: 0 12px;
	display: flex;
	align-items: center;
}

.title {
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
	margin-left: 8px;
	user-select: none;
}
