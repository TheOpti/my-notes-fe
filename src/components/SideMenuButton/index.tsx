import React from 'react';
import cx from 'classnames';

import Icon from 'components/Icon';

import styles from './styles.css';

type PropsType = {
	title: string;
	icon: string;
	active?: boolean;
	expanded: boolean;
	onClick: () => void;
};
const SideMenuButton: React.FC<PropsType> = (props: PropsType) => {
	const { title, icon, active, expanded, onClick } = props;

	const sideMenuBtnClasses = cx(styles.sideMenuButton, {
		[styles.active]: active,
		[styles.expanded]: expanded,
		[styles.collapsed]: !expanded,
	});

	return (
		<div className={sideMenuBtnClasses} onClick={onClick} data-testid="sidemenu-button">
			<Icon name={icon} size="medium" className={styles.icon} />
			<div className={styles.title} data-testid={`title-${expanded ? 'expanded' : 'collapsed'}`}>
				{title}
			</div>
		</div>
	);
};

SideMenuButton.defaultProps = {
	active: false,
};

export default SideMenuButton;
