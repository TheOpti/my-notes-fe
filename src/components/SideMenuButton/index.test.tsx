import React from 'react';
import { mount } from 'enzyme';
import SideMenuButton from './';

const noop = () => {};

describe('SideMenuButton', () => {
	it('should render correctly without any issues', () => {
		const wrapper = mount(
			<SideMenuButton title="Side menu title" icon="tag" active={false} expanded={false} onClick={noop} />
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should call onClickHandler function when SideMenuButton is actually clicked', () => {
		const mockFunction = jest.fn();

		const wrapper = mount(
			<SideMenuButton title="Side menu title" icon="tag" active={false} expanded={false} onClick={mockFunction} />
		);

		wrapper.find({ 'data-testid': 'sidemenu-button' }).simulate('click');
		expect(mockFunction).toHaveBeenCalled();
	});

	it('should not display text when button is not expanded', () => {
		const wrapper = mount(
			<SideMenuButton title="Side menu title" icon="tag" active={false} expanded={false} onClick={noop} />
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should correctly toggle title when button is clicked', () => {
		const wrapper = mount(
			<SideMenuButton title="Side menu title" icon="tag" active={false} expanded={false} onClick={noop} />
		);

		expect(wrapper.find({ 'data-testid': 'title-collapsed' }).html()).toEqual(
			'<div class="title" data-testid="title-collapsed">Side menu title</div>'
		);

		wrapper.setProps({ expanded: true });

		expect(wrapper.find({ 'data-testid': 'title-expanded' }).html()).toEqual(
			'<div class="title" data-testid="title-expanded">Side menu title</div>'
		);
	});
});
