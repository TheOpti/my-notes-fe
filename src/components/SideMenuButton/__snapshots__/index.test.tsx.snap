// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuButton should not display text when button is not expanded 1`] = `
<SideMenuButton
  active={false}
  expanded={false}
  icon="tag"
  onClick={[Function]}
  title="Side menu title"
>
  <div
    className="sideMenuButton collapsed"
    data-testid="sidemenu-button"
    onClick={[Function]}
  >
    <Icon
      className="icon"
      name="tag"
      size="medium"
    >
      <i
        className="root medium tag icon"
        data-testid="icon-tag"
      />
    </Icon>
    <div
      className="title"
      data-testid="title-collapsed"
    >
      Side menu title
    </div>
  </div>
</SideMenuButton>
`;

exports[`SideMenuButton should render correctly without any issues 1`] = `
<SideMenuButton
  active={false}
  expanded={false}
  icon="tag"
  onClick={[Function]}
  title="Side menu title"
>
  <div
    className="sideMenuButton collapsed"
    data-testid="sidemenu-button"
    onClick={[Function]}
  >
    <Icon
      className="icon"
      name="tag"
      size="medium"
    >
      <i
        className="root medium tag icon"
        data-testid="icon-tag"
      />
    </Icon>
    <div
      className="title"
      data-testid="title-collapsed"
    >
      Side menu title
    </div>
  </div>
</SideMenuButton>
`;
